sdks/java/harness/src/test/java/org/apache/beam/fn/harness/control/ProcessBundleHandlerTest.java

```diff
@@ -790,6 +811,367 @@ public void testPTransformStartExceptionsArePropagated() {
+        InstructionRequest.newBuilder()
+            .setInstructionId("998L")
+            .setProcessBundle(
+                ProcessBundleRequest.newBuilder()
+                    .setProcessBundleDescriptorId("1L")
+                    .setElements(
+                        Elements.newBuilder()
+                            .addData(
+                                Data.newBuilder()
+                                    .setInstructionId("998L")
+                                    .setTransformId("2L")
+                                    .setData(encodedData.toByteString())
+                                    .build())
+                            .addData(
+                                Data.newBuilder()
+                                    .setInstructionId("998L")
+                                    .setTransformId("2L")
+                                    .setIsLast(true)
+                                    .build())
+                            .addTimers(
+                                Timers.newBuilder()
+                                    .setInstructionId("998L")
+                                    .setTransformId("3L")
+                                    .setTimerFamilyId(
+                                        TimerFamilyDeclaration.PREFIX
+                                            + SimpleRecordingDoFn.TIMER_FAMILY_ID)
+                                    .setTimers(encodedTimer.toByteString())
+                                    .build())
+                            .addTimers(
+                                Timers.newBuilder()
+                                    .setInstructionId("998L")
+                                    .setTransformId("3L")
+                                    .setTimerFamilyId(
+                                        TimerFamilyDeclaration.PREFIX
+                                            + SimpleRecordingDoFn.TIMER_FAMILY_ID)
+                                    .setIsLast(true)
+                                    .build())
+                            .build()))
+            .build());
+    handler.shutdown();
+    assertThat(SimpleRecordingDoFn.consumedData, contains("data"));
+    assertThat(SimpleRecordingDoFn.firedOnTimerCallbackTimerIds, contains("timer_id"));
+    // Register timer family outbound receiver.
+    verify(beamFnDataClient).send(any(), any(), any());
+    verifyNoMoreInteractions(beamFnDataClient);
+  }
+
+  @Test
+  public void testInstructionEmbeddedElementsWithMalformedData() throws Exception {
+    ProcessBundleHandler handler = setupProcessBundleHanlderForSimpleRecordingDoFn();
+
+    ByteString.Output encodedData = ByteString.newOutput();
+    KvCoder.of(StringUtf8Coder.of(), StringUtf8Coder.of()).encode(KV.of("", "data"), encodedData);
+
+    assertThrows(
+        "Expect java.lang.IllegalStateException: Unable to find inbound data receiver for"
+            + " instruction 998L and transform 3L. But was not thrown or Exception did not match.",
```

Comment:
"But was not thrown or Exception did not match" ???

The failure cause makes it look like the wrong outcome was being met and not the one that was added specifically for this case.